File: 2460-apply-operations-to-an-array/2460-apply-operations-to-an-array.py
class Solution:
    def applyOperations(self, nums: List[int]) -> List[int]:
        left = 0 
        right = 1
        zeros = []
        while left < right and right < len(nums):
            if nums[left] != 0 and nums[left] == nums[right]:
                nums[left] *= 2
                nums[right] = 0
                left += 1
                right += 1
            right += 1
            left += 1

        non_zeros = [num for num in nums if num != 0]
        zero_count = len(nums) - len(non_zeros)
        nums = non_zeros + [0] * zero_count
        
        return nums

File: 2460-apply-operations-to-an-array/2460-apply-operations-to-an-array.go
func applyOperations(nums []int) []int {
	n := len(nums)
	for i := 0; i < n-1; i++ {
		if nums[i] == nums[i+1] && nums[i] != 0 {
			nums[i] *= 2
			nums[i+1] = 0
		}
	}

	j := 0
	for i := 0; i < n; i++ {
		if nums[i] != 0 {
			nums[j] = nums[i]
			j++
		}
	}

	for j < n {
		nums[j] = 0
		j++
	}

	return nums
}